ion/parse_text_simple.go

```diff
@@ -101,6 +130,21 @@ func doStringReplacements(str []byte) []byte {
 			case '\'', '"', '\\':
 				ret = append(ret, next)
 				index++
+			case 'x':
+				index += 2
+				data := p.decodeHex(str[index : index+2])
+				index += 1
+				ret = append(ret, data...)
+			case 'u':
+				index += 2
+				data := p.decodeHex(str[index : index+4])
+				index += 3
+				ret = append(ret, data...)
+			case 'U':
+				index += 2
+				data := p.decodeHex(str[index : index+8])
+				index += 7
+				ret = append(ret, data...)
```

Comment:
A lot of unchecked slice indexing.

Reply:
Refactoring this into `decodeHex` to have that do all the bounds checking.